packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,87 +15,73 @@ const IconDrawer = ({ className }) => (
     </svg>
 );
 
-class Drawer extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = { is_open: props.is_open };
-    }
+const Drawer = ({
+    anchor = 'left',
+    children,
+    className,
+    contentClassName,
+    footer,
+    header,
+    width = 250,
+    zIndex = 4,
+    ...props
```

Comment:
```suggestion
    is_open,
    toggleDrawer,
```

Reply:
I believe it is better to keep both `is_open` and `toggleDrawer` in props to avoid confusing.
because we have both toggleDrawer function and inside props value! same for `is_open`

Reply:
if we remove the is_open state, we can update props.is_open directly and in this case we can change the local `toggleDrawer` function name or directly using `toggleDrawer` props on the `onClick` handler. but I am just curios about `if` condition in the the local `toggleDrawer` method, in which case we do not have this `toggleDrawer` props?

Reply:
As i found in the code we may not pass the toggleDrawer function as a props to the Drawer component, so in this case we can not update is_open props directly and we should change it with local toggleDrawer method. so we can not remove the local method and we need is_open state as well.

Reply:
I agree with you